### PR TITLE
Kill the struct wrapper around Client

### PR DESCRIPTION
**Pull request by zenhack**
...in generated interface types. This means that Client and the wrapper
types have the same underlying type, so we can CapList for both.

Notes:

- The pogs package has some test failures that need fixing.
- Annoyingly, we can't just use ~Client as the constraint, since the
  proper underlying type is `struct { *client }`. But it works.

---

Marking this as a draft for now.

### CODE REVIEW

**Review by lthibault — DISMISSED** [review]

**Review by lthibault — APPROVED** [review]

**Comment by zenhack** [comment]
Fixed the tests. One thing I *really* don't like about this is that go doc shows the constraint as `~struct { *client }`, which is exposing implementation detail. And if you do `go doc Client`, it doesn't tell you that the underlying type is the same.

**Comment by lthibault** [comment]
Forgive me if we've covered this before, but is there a reason why the constraint can't be `~Client`?

**Comment by zenhack** [comment]
If I use `~Client`, I get:

```
./list.go:1151:16: invalid use of ~ (underlying type of Client is struct{*client})
```

**Comment by zenhack** [comment]
Per discussion via matrix, I pushed another commit that makes an alias `type ClientKind = ~struct { *client }`, which makes this a bit better.

File: pogs/pogs_test.go [hunk 1]
```diff
@@ -161,13 +161,13 @@ var goodTests = []Z{
...
 	{Which: air.Z_Which_echo, Echo: air.Echo{}},
-	{Which: air.Z_Which_echo, Echo: air.Echo{Client: capnp.ErrorClient(errors.New("boo"))}},
+	{Which: air.Z_Which_echo, Echo: air.Echo(capnp.ErrorClient(errors.New("boo")))},
 	{Which: air.Z_Which_echoes, Echoes: []air.Echo{
-		{Client: capnp.Client{}},
-		{Client: capnp.ErrorClient(errors.New("boo"))},
-		{Client: capnp.Client{}},
-		{Client: capnp.ErrorClient(errors.New("boo"))},
-		{Client: capnp.Client{}},
+		air.Echo{},
```
Comment (lthibault) [inline pogs/pogs_test.go:166]: Nit:  I think this can be simplified to `{}`.  Some linters complain about this, so I vote to reduce such noise.

**Review by lthibault — APPROVED** [review]
Minor issue inline, otherwise LGMT.

**Review by lthibault — APPROVED** [review]

**Pull request merged**